src/rl78.c

```diff
@@ -40,48 +48,105 @@ static void rl78_set_reset(port_handle_t fd, int mode, int value)
     {
         serial_set_dtr(fd, level);
     }
+    }
 }
 
 int rl78_reset_init(port_handle_t fd, int wait, int baud, int mode, float voltage)
+// Routine for resetting RL78 and initializing it in programming mode
 {
     unsigned char r;
+    //(mode = desired communication mode from terminal input - 1)
+
+   // Determine UART mode from mode argument
     if (MODE_UART_1 == (mode & MODE_UART))
     {
         r = SET_MODE_1WIRE_UART;
         communication_mode = 1;
     }
+    
     else
     {
         r = SET_MODE_2WIRE_UART;
         communication_mode = 2;
     }
     if (4 <= verbose_level)
     {
+        if (mode == 4) // Special case for mode 5
+        {
+            communication_mode = 5;
+            printf("Using communication mode 5 (HW RESET with CTS reading RESET)\n");
+        }
+        else
+        {
         printf("Using communication mode %u%s\n",
                (mode & (MODE_UART | MODE_RESET)) + 1,
                (mode & MODE_INVERT_RESET) ? " with RESET inversion" : "");
+        }   
     }
-    rl78_set_reset(fd, mode, 0);                            /* RESET -> 0 */
-    serial_set_txd(fd, 0);                                  /* TOOL0 -> 0 */
-    if (wait)
+    // Begin reset procedure
+    if (mode == 4) // sequencing for mode 5
```

Comment:
`if (MODE_RESET_INPUT_TRUE == (mode & MODE_RESET_INPUT))`

(ngl, I don't understand putting the constant before the == thing. But maintaining code style is more important than using MY code style.)